cpg-core/src/test/kotlin/de/fraunhofer/aisec/cpg/enhancements/EOGTest.kt

```diff
@@ -1203,6 +1203,78 @@ internal class EOGTest : BaseTest() {
         )
     }
 
+    @Test
+    fun testLambdaExpression() {
+        val config =
+            TranslationConfiguration.builder()
+                .sourceLocations(File("src/test/resources/cxx/lambdas.cpp"))
+                .defaultPasses()
+                .defaultLanguages()
+                .build()
+        val analyzer = TranslationManager.builder().config(config).build()
+        val result = analyzer.analyze().get()
+
+        assertNotNull(result)
+        val function = result.functions["lambda2"]
+        assertNotNull(function)
+
+        val lambdaVar = function.variables["this_is_a_lambda"]
+        assertNotNull(lambdaVar)
+        val lambda = lambdaVar.initializer as? LambdaExpression
+        assertNotNull(lambda)
+
+        // The "outer" EOG is assembled correctly.
+        assertTrue(lambda in lambdaVar.prevEOG)
+        val printFunctionCall = function.calls["print_function"]
+        assertNotNull(printFunctionCall)
+        assertTrue(printFunctionCall in lambda.prevEOG)
+
+        // The "inner" EOG is assembled correctly.
+        val body = (lambda.function?.body as? CompoundStatement)
+        assertNotNull(body)
+        assertEquals(1, lambda.function?.nextEOG?.size)
+        assertEquals(
+            "std::cout",
+            (lambda.function?.nextEOG?.get(0) as? DeclaredReferenceExpression)?.name.toString()
+        )
+
+        val cout = lambda.function?.nextEOG?.get(0) as? DeclaredReferenceExpression
+        assertEquals(1, cout?.nextEOG?.size)
+        assertEquals("Hello ", (cout?.nextEOG?.get(0) as? Literal<*>)?.value.toString())
+
+        val hello = cout?.nextEOG?.get(0) as? Literal<*>
+        assertEquals(1, hello?.nextEOG?.size)
```

Comment:
Inserting a couple of assertNotNull would be good to drastically remove the amount of `?` in the code